bench/micro/core/fiber/chain.rb

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+def create_chain(size)
+    last = Fiber.new { loop { Fiber.yield 42 } }
+    (size - 1).times.inject(last) { |nxt|
```

Comment:
`inject` also yields each element as a second block argument.
I am not exactly sure why it works here but it might be less surprising to use `{ |nxt, _|`.

Reply:
proc does not check args count

Reply:
Yes, but why does it not group arguments here like in
```ruby
{a:1}.each {|a| p a} # => [:a, 1]
```
Because in the implementation it's yield(acc, element) and not yield([acc, element]) like in Hash#each? I can't say I know all of these by heart.

Reply:
> Because in the implementation it's yield(acc, element) and not yield([acc, element]) like in Hash#each?

I think so

Reply:
Fixed.